tests/test_core.py

```diff
@@ -298,6 +298,44 @@ def test_300_restrict(self):
         self.assertNotEqual(aman.a1[0, 0, 0, 1], 0.)
 
     # wrap of AxisManager, merge.
+    def test_get_set(self):
+        dets = ["det0", "det1", "det2"]
+        n, ofs = 1000, 0
+        aman = core.AxisManager(
+            core.LabelAxis("dets", dets), core.OffsetAxis("samps", n, ofs)
+        )
+        child = core.AxisManager(
+            core.LabelAxis("dets", dets + ["det3"]),
+            core.OffsetAxis("samps", n, ofs - n // 2),
+        )
+
+        child2 = core.AxisManager(
+            core.LabelAxis("dets2", ["det4", "det5"]),
+            core.OffsetAxis("samps", n, ofs - n // 2),
+        )
+        aman.wrap("child", child)
+        aman["child"].wrap("child2", child2)
+        self.assertEqual(aman["child.child2.dets2"].count, 2)
+        self.assertEqual(aman["child.dets"].name, "dets")
+        np.testing.assert_array_equal(aman["child.child2.dets2"].vals, np.array(["det4", "det5"]))
+        with self.assertRaises(KeyError):
+            aman["child.someentry"]
+
+        with self.assertRaises(KeyError):
+            aman["child2"]
+
+        with self.assertRaises(AttributeError):
+            aman["child.dets.an_extra_layer"]
+
+        self.assertIn("child.dets", aman)
+        self.assertIn("child.dets2", aman) # I am not sure why this is true
+        self.assertNotIn("child.child2.someentry", aman)
+
+        aman["child"] = child2
```

Comment:
This is a weird and dangerous case -- because AxisManagers have axes, and those axes need to be kept in sync, it's odd to set this directly.  One should "always use wrap when adding an AxisManager to an AxisManager".  So I might remove this test, since it seems to legitimize unsafe usage.

Reply:
I will raise a ValueError when that is happening. Is there a valid case? If you see the diff in `__setitem__` it is not that different apart from going through the path.